core/src/main/scala/org/apache/spark/internal/config/UI.scala

```diff
@@ -97,6 +97,12 @@ private[spark] object UI {
     .booleanConf
     .createWithDefault(true)
 
+  val UI_FLAMEGRAPH_ENABLED = ConfigBuilder("spark.ui.threadDump.flamegraphEnabled")
+    .doc("Whether to render the Flamegraph for executor thread dumps")
```

Comment:
I think we should describe this happens within UI ;-)?

Reply:
Do you mean that we should add a short paragraph to describe the flame graph gone because users turned it off? I considered this, but it seems that we don't do such things for other UI elements or switches, i.e., timeline, heapdump.